src/Pages/UserPages/Landing/Landing.tsx

```diff
@@ -1,19 +1,292 @@
+import { useNavigate } from 'react-router-dom';
 const Landing = () => {
 
   const dispatch = useDispatch();
+  const { count } = useSelector((state) => state.AddToFavorite)
+  const { data } = useSelector((state) => state.RemoveFavoriteItemSlice)
+
+  const [personsCount, setPersonsCount] = useState(1);
+const navigate =  useNavigate()
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedDateRange, setSelectedDateRange] = useState<Range<Dayjs>>([
+    dayjs('2024-01-25'),
+    dayjs('2024-01-30'),
+  ]);
+console.log(selectedDateRange);
+console.log(personsCount);
+
 
   useEffect(() => {
+    getAdsData()
     dispatch(fetchDataIslogged());
+    getFavoriteData()
+  }, [dispatch,count,data]);
+
+
+  const handleCalendarChange = (newDateRange: Range<Dayjs>) => {
+    setSelectedDateRange(newDateRange);
+  };
+
+  const handleButtonClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handlePopoverClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleDateSelected = () => {
+    handlePopoverClose();
+  };
+
+  const handleIncrease = () => {
+    setPersonsCount(personsCount + 1);
+  };
+
+  const handleDecrease = () => {
+    if (personsCount > 1) {
+      setPersonsCount(personsCount - 1);
+    }
+  };
+
+
+const startDate =selectedDateRange[0]?.format('YYYY-MM-DD')
+const endDate =selectedDateRange[1]?.format('YYYY-MM-DD')
+
+  console.log(`${selectedDateRange[0]?.format('YYYY-MM-DD')} - ${selectedDateRange[1]?.format('YYYY-MM-DD')}`);
+  const [adsData, setAdsData] = useState();
+  //! ************************ Rooms Ads *************************
+  const getAdsData = useCallback(async () => {
+
+    try {
+      // @ts-ignore
+      const element = await dispatch(AllAdsData());
+      // @ts-ignore
+      setAdsData(element?.payload?.data?.data?.rooms);
+    } finally {
+
+    }
   }, [dispatch]);
 
+
+
+
+  //! ************************ Add To Favorite  *************************
+  const addItemToFavorite = useCallback(async (roomId: any) => {
+    try {
+      // @ts-ignore
+      const element = await dispatch(AddFavoriteItem(roomId));
+      console.log(element)
+      // @ts-ignore
+      toast.success(element?.payload?.message, {
+        autoClose: 2000,
+        theme: "colored",
+      });
+    } catch (error) {
+      // toast.error("Error fetching data:", error);
+    }
+  }, [dispatch]);
+
+  //! ************************ Get All Favorite Rooms  *************************
+  const [favList, setFavList] = useState([])
+  const getFavoriteData = useCallback(async () => {
+    try {
+      // @ts-ignore
+      const element = await dispatch(getFavoriteItems());
+      // @ts-ignore
+      setFavList(element?.payload?.data?.favoriteRooms[0]?.rooms)
+
+    } finally {
+
+    }
+  }, [dispatch]);
+
+
+  //! ************************ Delete From Favorite  *************************
+
+  const deleteFavoriteItem = useCallback(async (roomId: any) => {
```

Comment:
There is no benefit memoizing this function, as it is neither complicated nor we pass it to a child component.